src/Osma.Mobile.App/ViewModels/Connections/ConnectionViewModel.cs

```diff
@@ -74,13 +87,24 @@ public Task RefreshTransactions()
         #region Bindable Command
         public ICommand NavigateBackCommand => new Command(async () =>
         {
-            // TODO Do we need to check there is something to navigate to
             await NavigationService.PopModalAsync();
         });
 
         public ICommand DeleteConnectionCommand => new Command(async () =>
         {
-            // TODO OS-218 Delete the connection
+            var dialog = DialogService.Loading("Deleting");
+
+            var context = await _agentContextProvider.GetContextAsync();
+            await _connectionService.DeleteAsync(context, _record.Id);
+
+            _eventAggregator.Publish(new ApplicationEvent() { EventType = ApplicationEventType.ConnectionsUpdated });
```

Comment:
Thoughts on publishing the event from inside connectionService.DeleteAsync instead?

Reply:
The argument could be made that we need more events at crucial points emitted by the framework, @tmarkovski your thoughts about a more prolific usage of events inside the framework for cases such as this? In any case these would not be application events instead framework or service events. My personal thoughts are it doesnt feel too onerous for the application implementer having to write their own events over the framework and that it currently allows them to maintain good control over the event logic and structure.